run.py

```diff
@@ -21,11 +21,19 @@ def main():
 
     init_rabbitmq()  # test the connection to the rabbitmq cluster
 
-    subscriber_future = setup_subscription()
-    with Readiness(os.getenv('READINESS_FILE_PATH', os.path.join(os.getcwd(), 'pubsub-ready'))):  # Indicate ready after successful setup
-        while True:  # setup_subscription creates a background thread for processing messages
-            subscriber_future.result(timeout=None)  # block main thread while polling for messages indefinitely
-            time.sleep(30)
+    futures = [setup_subscription(),
+               setup_subscription(subscription_name=OFFLINE_SUBSCRIPTION_NAME, callback=offline_receipt_to_case,
+                                  subscription_project_id=OFFLINE_SUBSCRIPTION_PROJECT_ID)]
+    executor_futures = []
+
+    with Readiness(os.getenv('READINESS_FILE_PATH',
+                             os.path.join(os.getcwd(), 'pubsub-ready'))):  # Indicate ready after successful setup
+        with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
After investigating, I don't think we need to manually spin up other threads here to run `background_pubsub_task.result`, once you have called `setup_subscription` the google pubsub library has spawn the background worker threads itself. Instead I believe we just need some sort of loop periodically checking the future objects in `futures` to make sure those threads haven't died in the background.